### PR TITLE
Added mask_indices 

### PR DESCRIPTION
**Pull request by umbreeze**
- Added mask_indices function for numpy frontend
- Part of https://github.com/unifyai/ivy/issues/1536

### CODE REVIEW

**Comment by ivy-leaves** [comment]
If you are working on an open task, please **edit** the PR description to link to the issue you've created.

For more information, please check [ToDo List Issues Guide](https://unify.ai/docs/ivy/overview/contributing/the_basics.html#todo-list-issues).

Thank you :hugs:

**Comment by umbreeze** [comment]
The failed check seems to be unrelated to my changes.

**Comment by Infrared1029** [comment]
Great work @umbreeze, looks good for the most part, there is a bug that the tests did not catch, that is when the `mask` function is actually from `np_frontend`, e.g, `np_frontend.triu`, the function fails (because `ivy.array` and `ivy.argwhere` do not accept `np_frontend.ndarray` which the `mask` function will return).

A quick fix would be to check for the `ivy_array` attr from the resulting `mask` and doing `mask = mask.ivy_array` if it exists.

**Comment by umbreeze** [comment]
- Made the necessary changes with the recommended fix @Infrared1029  
- The failed checks seem to be unrelated to my changes.

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @umbreeze 

The tests are passing for other backends except for the Tensoflow backend,
Would you fix this🙂

Thanks

**Comment by umbreeze** [comment]
The test failures from the logs of the actions look like :

`Run find . -name "new_failures_*.txt" -exec cat {} + > combined_failures.txt
This PR introduces the following new failing tests:
ivy_tests/test_ivy/test_frontends/test_numpy/test_mathematical_functions/test_miscellaneous.py::test_numpy_clip,paddle
ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_elementwise.py::test_gradient,paddle
ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_statistical.py::test_quantile,torch
ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py::test_torch_instance_cross,numpy
ivy_tests/test_ivy/test_frontends/test_tensorflow/test_raw_ops.py::test_tensorflow_LeakyReLU,numpy
ivy_tests/test_ivy/test_frontends/test_paddle/test_nn/test_functional/test_paddle_loss.py::test_paddle_binary_cross_entropy_with_logits,torch
ivy_tests/test_ivy/test_functional/test_experimental/test_nn/test_losses.py::test_log_poisson_loss,torch
ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py::test_torch_instance_index_add_,paddle
ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tensor.py::test_tensorflow_instance_getitem,tensorflow
ivy_tests/test_ivy/test_frontends/test_jax/test_jax_lax_operators.py::test_jax_lax_gt,jax
ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_searching.py::test_unravel_index,numpy
ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_searching.py::test_unravel_index,torch`

@hirwa-nshuti Are these failures related to my changes? The tensorflow backend doesn't seem to be an issue locally. 

Thanks!

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Try to update your branch with the latest changes and run tests from your end then attach a screenshot of your test run

**Comment by umbreeze** [comment]
@hirwa-nshuti They seem to be passing with both numpy and tensorflow backends. This was after I updated my branch.

image
image

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @umbreeze 

I tried to run the tests locally and I'm getting this failure🙂
image

Would you look into it?

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by umbreeze** [comment]
Could I get some guidance regarding these failed checks? I've been through the hypothesis docs and the error seems to coming from my sampling filters/boundaries, but i am not aware of where these lint errors are coming from. @Infrared1029 @hirwa-nshuti

**Comment by umbreeze** [comment]
@hirwa-nshuti could you check if the sampling error is still coming up?

**Review by hirwa-nshuti — CHANGES_REQUESTED** [review]
Hey @umbreeze 

The tests are failing strangely for both the backends, would you update your branch with master branch changes and try to run the tests locally to see if they pass.

Thanks🙂
